src/rive/rive_render_private.cpp

```diff
@@ -19,11 +24,119 @@ namespace rive
         rgbaOut[3] = (float)((0xff000000 & colorIn) >> 24) / 255.0f;
     }
 
-    SharedRenderPaint::SharedRenderPaint()
-    : m_Builder(0)
+    ////////////////////////////////////////////////////////
+    // SharedRenderPaint
+    ////////////////////////////////////////////////////////
+
+    SharedRenderPaint::SharedRenderPaint(Context* ctx)
+    : m_Context(ctx)
+    , m_Builder(0)
+    , m_Stroke(0)
+    , m_StrokeBuffer(0)
     , m_Data({})
     {}
 
+    SharedRenderPaint::~SharedRenderPaint()
+    {
+        if (m_Builder != 0)
+        {
+            delete m_Builder;
+        }
+        if (m_Stroke != 0)
+        {
+            delete m_Builder;
+        }
+        if (m_StrokeBuffer)
+        {
+            m_Context->m_DestroyBufferCb(m_StrokeBuffer, m_Context->m_BufferCbUserData);
+        }
+    }
+
+
+    RenderPaintStyle SharedRenderPaint::getStyle()
+    {
+        return m_Style;
+    }
+
+    void SharedRenderPaint::thickness(float value)
+    {
+        m_StrokeThickness = value;
+    }
+    
+    void SharedRenderPaint::join(StrokeJoin value)
+    {
+        m_StrokeJoin = value;
+    }
+    
+    void SharedRenderPaint::cap(StrokeCap value)
+    {
+        m_StrokeCap = value;
+    }
+
+    void SharedRenderPaint::blendMode(BlendMode value)
+    {
+        // NOP
+    }
+
+    void SharedRenderPaint::style(RenderPaintStyle value)
+    {
+        m_Style = value;
+
+        if (m_Stroke != 0)
+        {
+            delete m_Stroke;
+        }
+
+        if (m_Style == RenderPaintStyle::stroke)
+        {
+            m_Stroke = new ContourStroke();
+            m_StrokeDirty = true;
+        }
+        else
+        {
+            m_Stroke = 0;
+            m_StrokeDirty = false;
+        }
+    }
+
+    void SharedRenderPaint::invalidateStroke()
+    {
+        if (m_Stroke != 0)
+        {
+            m_StrokeDirty = true;
+        }
+    }
+
+    void SharedRenderPaint::drawPaint(SharedRenderer* renderer, const Mat2D& transform, SharedRenderPath* path)
+    {
+        if (m_Stroke != 0)
+        {
+            if (m_StrokeDirty)
+            {
+                static Mat2D identity;
+                m_Stroke->reset();
+                path->extrudeStroke(m_Stroke,
+                                    m_StrokeJoin,
+                                    m_StrokeCap,
+                                    m_StrokeThickness / 2.0f,
+                                    identity);
+                m_StrokeDirty = false;
+            }
+            const std::vector<Vec2D>& strip = m_Stroke->triangleStrip();
+            size_t size = strip.size();
+            if (size == 0)
+            {
+                return;
+            }
+
+            m_StrokeBuffer = renderer->m_Context->m_RequestBufferCb(m_StrokeBuffer, BUFFER_TYPE_VERTEX_BUFFER,
+                (void*) &strip[0][0], size * 2 * sizeof(float), renderer->m_Context->m_BufferCbUserData);
```

Comment:
Not sure how many places the sizeof(Vertex) we have, but it might be worth considering to add one?